jobs/pipelines/powervm/zstreams/zstream-trigger-job/Jenkinsfile

```diff
@@ -0,0 +1,186 @@
+pipeline {
+    agent {
+        docker {
+            image 'quay.io/powercloud/inbound-agent:4.6-1.3'
+            args '-v /etc/resolv.conf:/etc/resolv.conf -v /var/run/docker.sock:/var/run/docker.sock'
+            label 'jump-vpc-x86_64'
+        }
+    }
+
+    stages {
+        stage('Pull Artifact OCP Builds') {
+            steps {
+                script {
+                    // Copy the builds of OCP releases from last successful job
+                    step([  $class: 'CopyArtifact', filter: "versions/*", fingerprintArtifacts: true, projectName: 'ZStream_Trigger', target: ".", optional: true, selector: lastSuccessful() ])
+                }
+            }
+        }
+            
+        stage("Identify newer Zstrems and Trigger Jenkins Jobs") {
+            steps {
+                script {
+                    def version_exists = "true"
+                    if(!fileExists("versions/")){
+                        version_exists = "false"
+                        echo "First Time Job Triggered! Initializing the versions!"
+                        echo "Not triggering any Jobs for first time!"
+                    }
+                    filenames = getChangedFilesList()
+                    if(filenames != null && version_exists == "true"){
+                        filenames = filenames.toSet()
+                        for (int i = 0; i < filenames.size(); i++) {
+                            filename        = filenames[i].trim()                      
+                            echo "Modified File: ${filenames[i]}"
+
+                            //Fetching release files information
+                            (release, next_release, next_release_file) = getReleases(filename)
+                            latestBuild   = readFile(filename).readLines()
+                            latestBuild   = latestBuild[0].trim()
+                            
+                            if (latestBuild != "" && latestBuild != null){
+                                currentStableBuild_url  = getStableBuild(latestBuild)
+                                nextStableBuild_url     = next_release_builds(next_release_file)
+                                echo "Triggering builds: ${latestBuild}, ${currentStableBuild_url}, ${nextStableBuild_url}"
+                                triggerBuild(release, latestBuild, currentStableBuild_url, nextStableBuild_url, next_release)
+                            }
+                        }
+                    }      
+                }
+            }
+        }
+    }
+
+    post {
+        always {
+            archiveArtifacts allowEmptyArchive: true, artifacts: "versions/*", fingerprint: true, onlyIfSuccessful: false
```

Comment:
@varad-ahirwadkar  Can we use here archiveAllArtifacts method?

Reply:
yes sure